server/notification-providers/wpush.js

```diff
@@ -0,0 +1,50 @@
+const NotificationProvider = require("./notification-provider");
+const axios = require("axios");
+const { DOWN, UP } = require("../../src/util");
+
+class WPush extends NotificationProvider {
+    name = "WPush";
+
+    /**
+     * @inheritdoc
+     */
+    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
+        const okMsg = "Sent Successfully.";
+
+        try {
+            const ret = await axios.post("https://api.wpush.cn/api/v1/send", {
+                "title": this.checkStatus(heartbeatJSON, monitorJSON),
+                "content": msg,
+                "apikey": notification.wpushAPIkey,
+                "channel": notification.wpushChannel
+            });
+            if (ret.data.code !== 0) {
+                return ret.data.message;
+            }
+
+            return okMsg;
+
+        } catch (error) {
+            this.throwGeneralAxiosError(error);
+        }
+    }
+
+    /**
+     * Get the formatted title for message
+     * @param {?object} heartbeatJSON Heartbeat details (For Up/Down only)
+     * @param {?object} monitorJSON Monitor details (For Up/Down only)
+     * @returns {string} Formatted title
+     */
+    checkStatus(heartbeatJSON, monitorJSON) {
```

Comment:
I think `checkStatus` does not really give an accurate meaning what this function does. What do you think about this instead?

```suggestion
    generateTitle(heartbeatJSON, monitorJSON) {
```